packages/driver/cypress/e2e/commands/querying/querying.cy.js

```diff
@@ -1020,6 +1020,11 @@ describe('src/cy/commands/querying', () => {
       })
     })
 
+    // https://github.com/cypress-io/cypress/issues/25225
+    it('returns one element when given multiple subjects with no children', () => {
+      cy.get('button').contains('submit').should('have.length', 1)
```

Comment:
I'd like to see test coverage from returning one element when multiple subjects regardless if the subjects have children or not. I glanced through these tests and not seeing an equivalent. When debugging this I didn't expect the issue to related to the subject's dom tree. Would always expect the first matching element

This might not be the right get selector - but I expect both scenarios to pass.
```suggestion
      cy.get('button').contains('submit').should('have.length', 1)
      cy.get('div').contains('submit').should('have.length', 1)
```

Reply:
I'll edit the test name to be a bit more descriptive - "has no children" isn't quite accurate. It's "when the subjects themselves match the selector, and none of their children do."

I think we already have lots of coverage around the case where .contains() simply matches multiple elements. Other tests in this file we have covering similar cases:

```
it('searches multiple subject elements', () => {
  cy.get('ul').contains('li', 'asdf 3')
})

it('resets the subject between chain invocations', () => {
  ...snip...
  cy.get('#complex-contains').contains('nested contains').then(($label) => {

it('reduces right by priority element', () => {
  ...snip...
  // it should find label because label is the first priority element
  // out of the collection of contains elements
  cy.get('#complex-contains').contains('nested contains')
```

Reply:
This returns the first matched element correct? This would align with our documentation

Reply:
> It's "when the subjects themselves match the selector, and none of their children do."

And do we have cases for these scenarios?
- the children match the selector and the subjects don't
- both the children and the subjects match the selector

Reply:
Its seems we've had a few holes when 12 went out with our test coverage, so Im surprised each of fixes has only had one test verifying its behavior. If we have existing tests for the above scenarios, mind linking them?

Reply:
> the children match the selector and the subjects don't

This is super common. For example,
```
<div>
  <span>Foo</span>
</div>

cy.get('div').contains('span', 'Foo')
```

The child matches the selector, but the parent does not. Some examples from the file:
```
it('searches multiple subject elements'
it('finds the furthest descendent when filter matches more than 1 element'
it('retries until it finds a filtered contains has the matching text node'
```

> both the children and the subjects match the selector

Every time a child matches, the parent also does.

```
<div>
  <div>Foo</div>
</div>

cy.get('div').contains('Foo')
```

The div contains the text 'Foo', and so does the span. Basically every test in the file covers this case.

Reply:
We are missing one in the form `.get('div').contains('div', 'text')`. I'll add that in a moment here.